File: bms/go/geo/geo_r2_Rect_String/geo_r2_Rect_String.go
package test

import (
	"fmt"
)

func (r geo_r2_Rect) String() string { return fmt.Sprintf("[Lo%s, Hi%s]", r.Lo(), r.Hi()) }

func (r geo_r2_Rect) Lo() geo_r2_Point {
	return geo_r2_Point{r.X.Lo, r.Y.Lo}
}

func (r geo_r2_Rect) Hi() geo_r2_Point {
	return geo_r2_Point{r.X.Hi, r.Y.Hi}
}

type geo_r2_Rect struct {
	X, Y geo_r1_Interval
}

type geo_r1_Interval struct {
	Lo, Hi float64
}

type geo_r2_Point struct {
	X, Y float64
}

File: translations_only/fail/go/geo/geo_r2_Rect_String/mistral/BaseRepair/geo_r2_Rect_String.rs

use std::fmt;

#[derive(Debug, Clone, Copy)]
struct GeoR1Interval {
    lo: f64,
    hi: f64,
}

impl GeoR1Interval {
    fn new(lo: f64, hi: f64) -> Self {
        GeoR1Interval { lo, hi }
    }
}

#[derive(Debug, Clone, Copy)]
struct GeoR2Point {
    x: f64,
    y: f64,
}

#[derive(Debug, Clone, Copy)]
struct GeoR2Rect {
    x: GeoR1Interval,
    y: GeoR1Interval,
}

impl GeoR2Rect {
    fn lo(&self) -> GeoR2Point {
        GeoR2Point {
            x: self.x.lo,
            y: self.y.lo,
        }
    }

    fn hi(&self) -> GeoR2Point {
        GeoR2Point {
            x: self.x.hi,
            y: self.y.hi,
        }
    }
}

impl fmt::Display for GeoR2Rect {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(
            f,
            "[Lo{{{:?}, {:?}}}, Hi{{{:?}, {:?}}}]",
            self.lo().x, self.lo().y, self.hi().x, self.hi().y
        )
    }
}

fn main() {
    let r = GeoR2Rect {
        x: GeoR1Interval::new(0.0, 10.0),
        y: GeoR1Interval::new(0.0, 20.0),
    };
    println!("{}", r);
}
